flytekit/models/execution.py

```diff
@@ -13,52 +20,130 @@
 from flytekit.models.node_execution import DynamicWorkflowNodeMetadata
 
 
+class SystemMetadata(_common_models.FlyteIdlEntity):
+    def __init__(self, execution_cluster: str):
+        self._execution_cluster = execution_cluster
+
+    @property
+    def execution_cluster(self) -> str:
+        return self._execution_cluster
+
+    def to_flyte_idl(self) -> flyteidl.admin.execution_pb2.SystemMetadata:
+        return _execution_pb2.SystemMetadata(execution_cluster=self.execution_cluster)
+
+    @classmethod
+    def from_flyte_idl(cls, pb2_object) -> SystemMetadata:
+        """
+        :param flyteidl.admin.execution_pb2.SystemMetadata pb2_object:
+        :return: SystemMetadata
+        """
+        return cls(
+            execution_cluster=pb2_object.execution_cluster,
+        )
+
+
 class ExecutionMetadata(_common_models.FlyteIdlEntity):
     class ExecutionMode(object):
         MANUAL = 0
         SCHEDULED = 1
         SYSTEM = 2
 
-    def __init__(self, mode, principal, nesting):
+    def __init__(
+        self,
+        mode: int,
+        principal: str,
+        nesting: int,
+        scheduled_at: datetime.datetime = None,
```

Comment:
```suggestion
        scheduled_at: typing.Optional[datetime.datetime] = None,
```
and the other three?

Reply:
done